addons/account/tests/common.py

```diff
@@ -279,49 +308,18 @@ def setup_company_data(cls, company_name, chart_template=None, **kwargs):
         }
 
     @classmethod
-    def setup_multi_currency_data(cls, default_values=None, rate2016=3.0, rate2017=2.0):
-        default_values = default_values or {}
-        foreign_currency = cls.env['res.currency'].create({
-            'name': 'Gold Coin',
-            'symbol': '☺',
-            'rounding': 0.001,
-            'position': 'after',
-            'currency_unit_label': 'Gold',
-            'currency_subunit_label': 'Silver',
-            **default_values,
-        })
-        rate1 = cls.env['res.currency.rate'].create({
-            'name': '2016-01-01',
-            'rate': rate2016,
-            'currency_id': foreign_currency.id,
-            'company_id': cls.env.company.id,
-        })
-        rate2 = cls.env['res.currency.rate'].create({
-            'name': '2017-01-01',
-            'rate': rate2017,
-            'currency_id': foreign_currency.id,
-            'company_id': cls.env.company.id,
-        })
-        return {
-            'currency': foreign_currency,
-            'rates': rate1 + rate2,
-        }
-
-    @classmethod
-    def _instantiate_basic_test_tax_group(cls, company=None, country=None):
-        company = company or cls.env.company
-        vals = {
-            'name': 'Test tax group',
-            'company_id': company.id,
-            'tax_receivable_account_id': cls.company_data['default_account_receivable'].sudo().copy({'company_id': company.id}).id,
-            'tax_payable_account_id': cls.company_data['default_account_payable'].sudo().copy({'company_id': company.id}).id,
-        }
-        if country:
-            vals['country_id'] = country.id
-        return cls.env['account.tax.group'].sudo().create(vals)
+    def copy_account(cls, account, default=None):
+        suffix_nb = 1
+        while True:
+            new_code = '%s.%s' % (account.code, suffix_nb)
+            if account.search_count([('company_id', '=', account.company_id.id), ('code', '=', new_code)]):
+                suffix_nb += 1
+            else:
+                return account.copy(default={'code': new_code, 'name': account.name, **(default or {})})
 
     @classmethod
     def setup_armageddon_tax(cls, tax_name, company_data):
+        cash_basis_transition_account = company_data['default_account_tax_sale'] and company_data['default_account_tax_sale'].copy()
```

Comment:
add the kwargs to this method and make sure test_account_move_in_***.py files setup this tax with `amount_type='purchase'`